keras_cv/visualization/plot_image_gallery.py

```diff
@@ -45,32 +47,23 @@ def plot_image_gallery(
         shuffle_files=True,
     )
 
-
-    def unpackage_tfds_inputs(inputs):
-        return inputs["image"]
-
-    train_ds = train_ds.map(unpackage_tfds_inputs)
-    train_ds = train_ds.apply(tf.data.experimental.dense_to_ragged_batch(16))
-
     keras_cv.visualization.plot_image_gallery(
-        next(iter(train_ds.take(1))),
+        train_ds,
         value_range=(0, 255),
         scale=3,
-        rows=2,
-        cols=2,
     )
     ```
 
     ![example gallery](https://i.imgur.com/r0ndse0.png)
 
     Args:
-        images: a Tensor or NumPy array containing images to show in the
+        images: a Tensor, `tf.data.Dataset` or NumPy array containing images to show in the
             gallery.
         value_range: value range of the images. Common examples include
             `(0, 255)` and `(0, 1)`.
-        rows: number of rows in the gallery to show.
-        cols: number of columns in the gallery to show.
         scale: how large to scale the images in the gallery
+        rows: (Optional) number of rows in the gallery to show. Required if inputs are unbatched.
```

Comment:
After chatting with the team a bit, and thinking about this code some more, I think we should keep rows/cols required and get rid of the logic to try to infer how many images should be shown.

It seems reasonable to ask users for a number of rows/cols, and it makes the code nice and overt on the caller's side.

It also dramatically simplifies how this can be implemented, because we don't need to try to infer batch sizes for different data types, and to add support for tf.data.Datasets, we can just use `tfds.as_numpy(dataset.take(rows*cols))` (after optionally unbatching the dataset 😄 

Sorry for the churn on this, and thanks for the awesome work!

Reply:
I'd prefer an automatic rows and cols, mostly to remove many useless lines of code from our examples.
This can be implemented in just a few lines, even with a fancy aspect ratio adaptation:

```
N = len(images)
sqrtN = int(math.ceil(math.sqrt(N)))
aspect = sum([im.shape[1]/im.shape[0] for im in images]) / N # mean aspect ratio of images
fig = plt.figure(figsize=(15,15/aspect), frameon=False)
```

(I've been using this code in all my samples from the past couple of years)
And it's fine to require either both rows and cols or neither and not handle the fancy cases in between.